airsonic-main/src/test/java/org/airsonic/player/service/search/SearchServiceSpecialGenreTestCase.java

```diff
@@ -0,0 +1,242 @@
+ */
+public class SearchServiceSpecialGenreTestCase {
+
+    @ClassRule
+    public static final SpringClassRule classRule = new SpringClassRule() {
+        HomeRule homeRule = new HomeRule();
+
+        @Override
+        public Statement apply(Statement base, Description description) {
+            Statement spring = super.apply(base, description);
+            return homeRule.apply(spring, description);
+        }
+    };
+
+    @Rule
+    public final SpringMethodRule springMethodRule = new SpringMethodRule();
+
+    @Autowired
+    private MediaScannerService mediaScannerService;
+
+    @Autowired
+    private MusicFolderDao musicFolderDao;
+
+    @Autowired
+    private DaoHelper daoHelper;
+
+    @Autowired
+    private SearchService searchService;
+
+    @Autowired
+    private SettingsService settingsService;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Autowired
+    ResourceLoader resourceLoader;
+
+    @Before
+    public void setup() throws Exception {
+        populateDatabase();
+    }
+
+    private static boolean dataBasePopulated;
+
+    private static Function<String, String> resolvePath = (childPath) ->{
+        return MusicFolderTestData.resolveBaseMediaPath() + childPath;
+    };
+    
+    private List<MusicFolder> musicFolders;
+    
+    private List<MusicFolder> getTestMusicFolders() {
+        if (isEmpty(musicFolders)) {
+            musicFolders = new ArrayList<>();
+
+            File musicDir = new File(resolvePath.apply("Search/SpecialGenre"));
+            musicFolders.add(new MusicFolder(1, musicDir, "accessible", true, new Date()));
+
+        }
+        return musicFolders;
+    }
+
+    private int count = 1;
+
+    private synchronized void populateDatabase() {
+
+        try {
+            Thread.sleep(300 * count++);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        if (!dataBasePopulated) {
+            getTestMusicFolders().forEach(musicFolderDao::createMusicFolder);
+            settingsService.clearMusicFolderCache();
+            TestCaseUtils.execScan(mediaScannerService);
+            System.out.println("--- Report of records count per table ---");
+            Map<String, Integer> records = TestCaseUtils.recordsInAllTables(daoHelper);
+            records.keySet().stream().filter(s -> s.equals("MEDIA_FILE") // 20
+                    | s.equals("ARTIST") // 5
+                    | s.equals("MUSIC_FOLDER")// 3
+                    | s.equals("ALBUM"))// 5
+                    .forEach(tableName -> System.out
+                            .println("\t" + tableName + " : " + records.get(tableName).toString()));
+            System.out.println("--- *********************** ---");
+            dataBasePopulated = true;
+        }
+    }
+
+    @Test
+    public void testQueryEscapeRequires() {
+    
+        /*
+         * Legacy is not searchable. (FILE01 - FILE16)
```

Comment:
I don't understand this comment :/

Reply:
surely.

 - There are 19 files in ``src/test/resources/MEDIAS/Search/SpecialGenre/ARTIST1/ALBUM_A``.
 -  In FILE01 to FILE16, Special strings for Lucene syntax are stored as tag values ​​of Genre.

I need a post-update comparison for future searches,
but there is no case because legacy does not handle this at all.

-> I should have noted down the failure cases. I will add.

Reply:
When the index created by executing SearchServiceSpecialGenreTestCase is confirmed in the viewer, it is as follows.

![image](https://user-images.githubusercontent.com/27724847/60919113-dd863180-a2cf-11e9-996b-0106ebb7ae30.png)

Legacy analyzes Genre with StandardAnalyzer, so some strings are discarded.
As a result, no Genre field is created in the index.
Asserts to confirm this implementation.